### ISSUE TITLE
Spanner transaction.commit() didn't resolve/reject promise

### ISSUE THREAD

**Issue by eric4545** [googleapis/nodejs-spanner#217]
#### Environment details

  - OS: macOS 10.13.4 / 
  - Node.js version: 8.11.1
  - npm version: 5.6.0
  - `@google-cloud/spanner` version: 1.4.3

#### Steps to reproduce

  1. ???
  2. ???

Sometimes `await transcation.commit()` didnt resolve/reject promise over 120s. But the error not happen everytime as we cant reproduce the error, but when error happened we can see `Error`[1] of `Operations per second` in GCP console

[1] ![image](https://user-images.githubusercontent.com/6886275/41212615-cf04cbd8-6d71-11e8-832c-fd4127221cf5.png)

Code example may upon request .

Thanks!

**Comment by callmehiphop** [comment]
Hi @eric4545, would you be able to provide some code that reproduces this issue?

**Comment by walshie4** [comment]
#116 & #205 potentially related

**Comment by eric4545** [comment]
hi @callmehiphop, i still trying to create a example to reproduce the problem, but so far i cant reproduce the same error in our project. Is there any internal fix of the spanner?

**Comment by eric4545** [comment]
![image](https://user-images.githubusercontent.com/6886275/41709637-41948e2a-7576-11e8-8ff0-4ea63f07556c.png)
Just find this maybe is related?

**Comment by eric4545** [comment]
@walshie4 I think the cause is we promisifiy the `runTranscation`, we thought it is a callback function, but now we removed the promisify for `runTransaction` and will going to test it more.

And yes we can reproduce when running the script. As there are more than one `runTransaction` and `commit` together, and `runTranscation` will retry exec `runFn` if some error occurs, but in the script we promisified it when in the next retry it will just call resolve again so nothing will exec.
